src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -113,7 +114,7 @@ def calculate_invalidation_digest(requirements: Iterable[str]) -> str:
     """Returns an invalidation digest for the given requirements."""
     m = hashlib.sha256()
     inputs = {
-        "requirements": list(requirements),
+        "requirements": sorted(FrozenOrderedSet(requirements)),
```

Comment:
I think we might be able to safely leave this as a `set()` and save the import: `sorted` takes care of the ordering for us.

Reply:
I should comment why I did this: I wanted to avoid `set` because it would break if the input was already ordered. Python uses timsort https://en.wikipedia.org/wiki/Timsort, which is O(n) in the best case if it's already sorted, and O(n logn) in the worst case.

Reply:
two points:

1. set is guaranteed to be ordered as of Python 3.6, so running `sorted` on a set is equivalent to running it on a list.
2. `n` is quite small for requirements, so if (1) weren't true, it probably wouldn't matter

Reply:
🤦  fine :)